pennylane_forest/numpy_wavefunction.py

```diff
@@ -90,25 +90,44 @@ def ev(self, A, wires):
           float: expectation value :math:`\left\langle{A}\right\rangle = \left\langle{\psi}\mid A\mid{\psi}\right\rangle`
         """
         # Expand the Hermitian observable over the entire subsystem
-        A = self.expand_one(A, wires)
+        A = self.expand(A, wires)
         return np.vdot(self.state, A @ self.state).real
 
-    def expand_one(self, U, wires):
-        r"""Expand a one-qubit operator into a full system operator.
+    def expand(self, U, wires):
+        r"""Expand a multi-qubit operator into a full system operator.
 
         Args:
-          U (array): :math:`2\times 2` matrix
-          wires (Sequence[int]): target subsystem
+          U (array): :math:`2^n \times 2^n` matrix where n = wires.
+          wires (Sequence[int]): Target subsystems (order matters!)
 
         Returns:
-          array: :math:`2^n\times 2^n` matrix
+          array: :math:`2^N\times 2^N` matrix. The full system operator.
         """
-        if U.shape != (2, 2):
-            raise ValueError('2x2 matrix required.')
-        if len(wires) != 1:
-            raise ValueError('One target subsystem required.')
-        wires = wires[0]
-        before = 2**wires
-        after = 2**(self.num_wires-wires-1)
-        U = np.kron(np.kron(np.eye(before), U), np.eye(after))
-        return U
+        if self.num_wires == 1:
+            # total number of wires is 1, simply return the matrix
+            return U
+
+        N = self.num_wires
+        wires = np.asarray(wires)
+
+        if np.any(wires < 0) or np.any(wires >= N) or len(set(wires)) != len(wires):
+            raise ValueError('Bad target subsystems.')
+
+        # generate N qubit basis states via the cartesian product
```

Comment:
Did you mean `tensor product` instead of `cartesian product`? (See e.g. https://physics.stackexchange.com/questions/350508/tensor-product-vs-cartesian-product-for-composite-quantum-systems)

Reply:
Ah, I meant it more in the programmatic sense, in that `itertools.product` performs the cartesian product

Reply:
Ok, that makes sense.